test/pkg/ptptesthelper/ptptesthelper.go

```diff
@@ -272,6 +272,61 @@ func CheckSlaveSyncWithMaster(fullConfig testconfig.TestConfig) {
 	Expect(slaveMasterID).Should(HavePrefix(masterID), "Error match MasterID with the SlaveID. Slave connected to another Master")
 }
 
+func CreatePtpTestPrivilegedDaemonSet(daemonsetName, daemonsetNamespace, daemonsetContainerName string) *corev1.PodList {
+	const (
+		imageWithVersion = "quay.io/testnetworkfunction/debug-partner:latest"
+	)
+	// Create the client of Priviledged Daemonset
+	k8sPriviledgedDs.SetDaemonSetClient(client.Client.Interface)
+	// 1. create a daemon set for the node reboot
+	dummyLabels := map[string]string{}
+	daemonSetRunningPods, err := k8sPriviledgedDs.CreateDaemonSet(daemonsetName, daemonsetNamespace, daemonsetContainerName, imageWithVersion, dummyLabels, pkg.TimeoutIn5Minutes)
+
+	if err != nil {
+		logrus.Errorf("error : +%v\n", err.Error())
+	}
+	return daemonSetRunningPods
+}
+
+func RecoverySlaveNetworkOutage(fullConfig testconfig.TestConfig) {
+	logrus.Info("Recovery network outage begins ...........")
+
+	// Get a slave pod
+	slavePod, err := ptphelper.GetPTPPodWithPTPConfig((*ptpv1.PtpConfig)(fullConfig.DiscoveredClockUnderTestPtpConfig))
+	if err != nil {
+		logrus.Error("Could not determine ptp daemon pod selected by ptpconfig")
+	}
+	slavePodNodeName := slavePod.Spec.NodeName
+	logrus.Info("slave node name is", slavePodNodeName)
+
+	// Get the pod from ptp test daemonset set on the slave node
+	networkRecoveryDaemonSetRunningPods := CreatePtpTestPrivilegedDaemonSet(pkg.RecoveryNetworkOutageDaemonSetName, pkg.RecoveryNetworkOutageDaemonSetNamespace, pkg.RecoveryNetworkOutageDaemonSetContainerName)
```

Comment:
Should we also remove the daemonset after we are done running the test?

Reply:
Yes, done!